From: "K. Y. Srinivasan" <kys@microsoft.com>
Date: Sun, 12 Jan 2014 11:09:14 -0800
Patch-mainline: v3.14-rc1
Subject: Input: hyperv-keyboard - pass through 0xE1 prefix
Git-commit: c3c4d99485ea51cd354ed3cd955a8310703456b6
References: fate#317533

Pass through the 0xE1 prefix so atkbd can properly parse the scancode
data.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
Acked-by: <ohering@suse.de>
---
 drivers/input/serio/hyperv-keyboard.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/serio/hyperv-keyboard.c b/drivers/input/serio/hyperv-keyboard.c
index 3a83c3c..6132619 100644
--- a/drivers/input/serio/hyperv-keyboard.c
+++ b/drivers/input/serio/hyperv-keyboard.c
@@ -160,7 +160,9 @@ static void hv_kbd_on_receive(struct hv_device *hv_dev,
 			if (info & IS_E0)
 				serio_interrupt(kbd_dev->hv_serio,
 						XTKBD_EMUL0, 0);
-
+			if (info & IS_E1)
+				serio_interrupt(kbd_dev->hv_serio,
+						XTKBD_EMUL1, 0);
 			scan_code = __le16_to_cpu(ks_msg->make_code);
 			if (info & IS_BREAK)
 				scan_code |= XTKBD_RELEASE;
